[FIX] runbot: addapt 18.0

Xavier-Do opened the pull request.

Xavier-Do committed:
[FIX] runbot: addapt requirements

Xavier-Do committed:
[FIX] runbot: adapt slug usage

Xavier-Do committed:
[FIX] runbot: adapt cron

Xavier-Do committed:
[FIX] runbot: fix tests

Xavier-Do committed:
[FIX] runbot: remove unaccent

see https://github.com/odoo/odoo/pull/139568

Xavier-Do committed:
[FIX] runbot: fix query test

Williambraecky commented on runbot/models/branch.py:
```diff
@@ -135,7 +135,7 @@ def _compute_branch_url(self):
                     branch.branch_url = "https://%s/pull/%s" % (branch.remote_id.base_url, branch.name)
                 else:
-                    branch.branch_url = "https://%s/tree/%s" % (branch.remote_id.base_url, branch.name)
+                    branch.branch_url = "https://%s/list/%s" % (branch.remote_id.base_url, branch.name)
```
```suggestion
                    branch.branch_url = "https://%s/tree/%s" % (branch.remote_id.base_url, branch.name)
```

Williambraecky commented on runbot/templates/badge.xml:
```diff
@@ -3,7 +3,7 @@
     <template id="runbot.badge_default">&lt;?xml version="1.0"?&gt;
       <svg t-attf-xmlns="http://www.w3.org/2000/svg" t-attf-width="{{left.width + right.width}}" height="18">
-        <!-- from https://github.com/badges/shields/tree/master/templates -->
+        <!-- from https://github.com/badges/shields/list/master/templates -->
```
The link seems completely dead, it seems they don't provide 'templates' anymore?
I didn't check for too long though, perhaps we could change it to: 
```suggestion
        <!-- inspired by https://github.com/badges/shields/ -->
```

Xavier-Do replied:
We may check that, but for now I would just revert the invalid changes made by upgrade_code

```suggestion
        <!-- from https://github.com/badges/shields/list/master/templates -->
        <!-- from https://github.com/badges/shields/tree/master/templates -->
```

Williambraecky commented on runbot/tests/test_branch.py:
```diff
@@ -6,7 +6,7 @@
 
     def test_base_fields(self):
-        self.assertEqual(self.branch_server.branch_url, 'https://example.com/base/server/tree/master')
+        self.assertEqual(self.branch_server.branch_url, 'https://example.com/base/server/list/master')
```
```suggestion
        self.assertEqual(self.branch_server.branch_url, 'https://example.com/base/server/tree/master')
```

Xavier-Do committed:
[FIX] runbot: revert some upgrade_code errors

Co-authored-by: William Braeckman (wbr) <wbr@odoo.com>

Xavier-Do committed:
[IMP] runbot: apply upgrade code

Williambraecky committed:
[REF] runbot: replace t-esc with t-out

t-esc has been deprecated and uses redirects to t-out anyways, removing
since in dev mode it logs a warning in the terminal.

See odoo/odoo#81024

Williambraecky committed:
[FIX] runbot: replace old BS4 class with BS5

text-left -> text-start
remove data-toggle css rules
font-italic -> fst-italic

Williambraecky committed:
[FIX] runbot: fix frontend urls to backend

Urls are broken after 18.0 upgrade

Williambraecky committed:
[FIX] runbot: restore form views

Since https://github.com/odoo/odoo/pull/154710 form views can use
<chatter/> directly to include the chatter, not doing so completely
breaks the dom generated by the form view.
